kotlin/src/com/here/flexiblepolyline/FlexiblePolylineTest.kt

```diff
@@ -0,0 +1,380 @@
+            if (args.size > 0) {
+                lineLength = args[0].toInt()
+            }
+            val test = FlexiblePolylineTest()
+            test.testInvalidCoordinates()
+            test.testInvalidThirdDimension()
+            test.testConvertValue()
+            test.testSimpleLatLngEncoding()
+            test.testComplexLatLngEncoding()
+            test.testLatLngZEncode()
+            test.encodingSmokeTest()
+
+            //Decode test
+            test.testInvalidEncoderInput()
+            test.testThirdDimension()
+            test.testDecodeConvertValue()
+            test.testSimpleLatLngDecoding()
+            test.testComplexLatLngDecoding()
+            test.testLatLngZDecode()
+            test.decodingSmokeTest()
+            test.testVeryLongLine(lineLength)
+        }
+    }
+}
+
+private data class TestData(
+    val precision: Int = 0,
+    val thirdDimensionPrecision: Int = 0,
+    val thirdDimension: FlexiblePolyline.ThirdDimension? = null,
+    val latLngZs: List<FlexiblePolyline.LatLngZ>? = null
+){}
+
+private data class TestCase(
+    val testInput: String,
+    val testResult: String
+){}
+
+private class TestCaseReader(testInputFile: String, testResultFile: String) : Iterator<TestCase> {
+    private var totalLines = 0
+    private var currentLine = 0
+    private var testCases = mutableListOf<TestCase>()
+
+    init {
+        try {
+            Files.newBufferedReader(Paths.get(FlexiblePolylineTest.TEST_FILES_RELATIVE_PATH + testInputFile)).use { input ->
+                Files.newBufferedReader(Paths.get(FlexiblePolylineTest.TEST_FILES_RELATIVE_PATH + testResultFile)).use { result ->
+                    // read line by line and validate the test
+                    while (true) {
+                        val regex = "\\s|\\(|\\)".toRegex()
+                        val testInputFileLine = input.readLine();
+                        val testResultFileLine = result.readLine();
+
+                        if (testInputFileLine != null && testInputFileLine.isNotBlank() && testResultFileLine != null && testResultFileLine.isNotBlank()) {
+                            testCases.add(
+                                TestCase(
+                                    testInput = testInputFileLine.replace(regex, ""),
+                                    testResult = testResultFileLine.replace(regex, "")
+                                )
+                            )
+                            totalLines++
+                        } else {
+                            break
+                        }
+                    }
+                }
+            }
+        } catch (e: Exception) {
+            e.printStackTrace()
+            System.err.format("TestCaseReader - exception reading test case $testInputFile and $testResultFile at LineNo: $totalLines: %s%n", e)
+            throw RuntimeException("Test failed, as test data could not be loaded by TestCaseReader")
```

Comment:
On second thought, we don't really need exception handling which include the line information since the only exceptions that can occur would be caused by the file readers and those are not specific to the line content.